openslides/motions/access_permissions.py

```diff
@@ -34,7 +31,9 @@ def get_restricted_data(self, full_data, user):
             data = full_data
         else:
             data = full_data.copy()
-            for i, field in enumerate(self.get_comments_config_fields()):
+            if not isinstance(config['motions_comments'], list) or not all(isinstance(c, dict) for c in config['motions_comments']):
+                raise ValidationError({'detail': 'motions_comments has to be a list of dictionaries.'})
```

Comment:
This method should not run any validation but remove irregular fields from output or set output completely to None.